File: SDKs/amazonaws.com/appflow/2020-08-23/java/lib/src/main/java/openapisdk/models/shared/DestinationConnectorProperties.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * DestinationConnectorProperties
 *  This stores the information that is required to query a particular connector. 
**/
public class DestinationConnectorProperties {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("CustomerProfiles")
    public CustomerProfilesDestinationProperties customerProfiles;
    public DestinationConnectorProperties withCustomerProfiles(CustomerProfilesDestinationProperties customerProfiles) {
        this.customerProfiles = customerProfiles;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("EventBridge")
    public EventBridgeDestinationProperties eventBridge;
    public DestinationConnectorProperties withEventBridge(EventBridgeDestinationProperties eventBridge) {
        this.eventBridge = eventBridge;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Honeycode")
    public HoneycodeDestinationProperties honeycode;
    public DestinationConnectorProperties withHoneycode(HoneycodeDestinationProperties honeycode) {
        this.honeycode = honeycode;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("LookoutMetrics")
    public java.util.Map<String, Object> lookoutMetrics;
    public DestinationConnectorProperties withLookoutMetrics(java.util.Map<String, Object> lookoutMetrics) {
        this.lookoutMetrics = lookoutMetrics;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Redshift")
    public RedshiftDestinationProperties redshift;
    public DestinationConnectorProperties withRedshift(RedshiftDestinationProperties redshift) {
        this.redshift = redshift;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("S3")
    public S3DestinationProperties s3;
    public DestinationConnectorProperties withS3(S3DestinationProperties s3) {
        this.s3 = s3;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Salesforce")
    public SalesforceDestinationProperties salesforce;
    public DestinationConnectorProperties withSalesforce(SalesforceDestinationProperties salesforce) {
        this.salesforce = salesforce;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Snowflake")
    public SnowflakeDestinationProperties snowflake;
    public DestinationConnectorProperties withSnowflake(SnowflakeDestinationProperties snowflake) {
        this.snowflake = snowflake;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Upsolver")
    public UpsolverDestinationProperties upsolver;
    public DestinationConnectorProperties withUpsolver(UpsolverDestinationProperties upsolver) {
        this.upsolver = upsolver;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Zendesk")
    public ZendeskDestinationProperties zendesk;
    public DestinationConnectorProperties withZendesk(ZendeskDestinationProperties zendesk) {
        this.zendesk = zendesk;
        return this;
    }
}

File: SDKs/amazonaws.com/appflow/2020-08-23/python/src/sdk/models/shared/destinationconnectorproperties.py
import dataclasses
from typing import Any,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import customerprofilesdestinationproperties as shared_customerprofilesdestinationproperties
from ..shared import eventbridgedestinationproperties as shared_eventbridgedestinationproperties
from ..shared import honeycodedestinationproperties as shared_honeycodedestinationproperties
from ..shared import redshiftdestinationproperties as shared_redshiftdestinationproperties
from ..shared import s3destinationproperties as shared_s3destinationproperties
from ..shared import salesforcedestinationproperties as shared_salesforcedestinationproperties
from ..shared import snowflakedestinationproperties as shared_snowflakedestinationproperties
from ..shared import upsolverdestinationproperties as shared_upsolverdestinationproperties
from ..shared import zendeskdestinationproperties as shared_zendeskdestinationproperties


@dataclass_json
@dataclasses.dataclass
class DestinationConnectorProperties:
    r"""DestinationConnectorProperties
     This stores the information that is required to query a particular connector. 
    """
    
    customer_profiles: Optional[shared_customerprofilesdestinationproperties.CustomerProfilesDestinationProperties] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('CustomerProfiles') }})
    event_bridge: Optional[shared_eventbridgedestinationproperties.EventBridgeDestinationProperties] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('EventBridge') }})
    honeycode: Optional[shared_honeycodedestinationproperties.HoneycodeDestinationProperties] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Honeycode') }})
    lookout_metrics: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('LookoutMetrics') }})
    redshift: Optional[shared_redshiftdestinationproperties.RedshiftDestinationProperties] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Redshift') }})
    s3: Optional[shared_s3destinationproperties.S3DestinationProperties] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('S3') }})
    salesforce: Optional[shared_salesforcedestinationproperties.SalesforceDestinationProperties] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Salesforce') }})
    snowflake: Optional[shared_snowflakedestinationproperties.SnowflakeDestinationProperties] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Snowflake') }})
    upsolver: Optional[shared_upsolverdestinationproperties.UpsolverDestinationProperties] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Upsolver') }})
    zendesk: Optional[shared_zendeskdestinationproperties.ZendeskDestinationProperties] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Zendesk') }})
    
